torch/csrc/inductor/aoti_torch/shim_common.cpp

```diff
@@ -182,6 +185,186 @@ AOTITorchError aoti_torch_create_tensor_from_blob(
   });
 }
 
+static AOTITorchError aoti_torch__scaled_dot_product_flash_attention_internal(
+    AtenTensorHandle* ret0, // returns new reference
+    AtenTensorHandle* ret1, // returns new reference
+    AtenTensorHandle* ret2, // returns new reference
+    AtenTensorHandle* ret3, // returns new reference
+    int64_t* ret4,
+    int64_t* ret5,
+    AtenTensorHandle* ret6, // returns new reference
+    AtenTensorHandle* ret7, // returns new reference
+    AtenTensorHandle* ret8, // returns new reference
+    AtenTensorHandle query,
+    AtenTensorHandle key,
+    AtenTensorHandle value,
+    double dropout_p = 0.0,
+    bool is_causal = false,
+    bool return_debug_mask = false,
+    c10::optional<double> scale = c10::nullopt) {
```

Comment:
Is it supposed to be C friendly API? Why `c10::optional` then?

Reply:
> Is it supposed to be C friendly API? Why `c10::optional` then?

We provide C interface. The implementation can be in C++.